### PR TITLE
fix: authority parsing

### PR DESCRIPTION
**Pull request by marbemac**
The `parseAuthority` hook used to run after the lookup attempt for a pointer, e.g. `#/foo`. If the target document returned from the authority was not json already, this lookup would fail. This PR makes it so that parseAuthority runs right after the authority target is fetched, before lookup, fixing the problem.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,37 +33,35 @@
...
     "dependency-graph": "0.8.x",
     "fast-memoize": "2.x.x",
-    "immer": "^3.1.2",
+    "immer": "3.x.x",
     "lodash": "4.x.x",
     "urijs": "1.x.x"
   },
   "devDependencies": {
-    "@stoplight/scripts": "5.1.0",
+    "@stoplight/scripts": "7.0.0",
```
Comment (P0lip) [inline package.json]: Make sure everything important is exported in the main entry point.

File: src/runner.ts [hunk 2]
```diff
@@ -380,46 +396,13 @@ export class ResolveRunner implements Types.IResolveRunner {
...
-            // if (parsed.errors) {
-            // TODO: as mentioned above, allow caller to add errors/validations
-            // }
-
-            lookupResult.resolved.result = parsed.result;
-          } catch (e) {
-            // could not parse... roll back to original value
-            lookupResult.resolved.result = val;
-
-            lookupResult.error = {
```
Comment (P0lip) [inline src/runner.ts]: Shouldn't we leave this error out?

Comment (marbemac) [reply src/runner.ts]: The error is still captured as part of the `RESOLVE_AUTHORITY` error (see change in tests related to this).

Comment (P0lip) [reply src/runner.ts]: Yeah I saw the test was updated, but what if the parseAuthority function throws? Shouldn't we set the former error?

**Pull request merged**